utils/mgmt_system.py

```diff
@@ -71,6 +88,19 @@ def get_search_filter_spec(self, *args, **kwargs):
 
         return pfs
 
+    def __getattribute__(self, attr):
+        # fetch the attribute using parent class to avoid recursion
+        res = super(_PsphereClient, self).__getattribute__(attr)
+        # any callable (except 'login') is protected against unexpected logout
+        if callable(res) and attr != 'login':
+            try:
+                return self._cached_retry[attr]
```

Comment:
The traceback is not helpful in PRT, but I suspect this is where recursion as happening, since `self._cached_retry `means a call to `self.__getattribute__('_cached_retry')`

Maybe just tweak the conditional to allow for that?
```python
if attr not in ('login', '_cached_retry') and callable(res):
```

Also, LBYL might be preferable to EAFP. This pattern has always seemed a little cleaner to me:
```python
if attr not in self._cached_retry:
    self._cached_retry[attr] = login_retry_wrapper(self, res)
return self._cached_retry[attr]
```